lib/crawler.js

```diff
@@ -359,71 +431,173 @@ Crawler.prototype._buildHttpRequest = function _buildHTTPRequest (options) {
 	}
 };
 
-Crawler.prototype._onContent = function _onContent (error, options, response) {
+Crawler.prototype._buildHttp2Session = function _buildHttp2Session(targetHost) {
+	const self = this;
+
+	self.http2Connections[targetHost] = http2.connect(targetHost);
+
+	log('debug', `connect to a new ${targetHost}`);
+	self.http2Connections[targetHost].on('error', (err) => {
+		log('warn', `Http2 stession error ${targetHost}, got error ${err}`);
+	});
+
+	self.http2Connections[targetHost].on('goaway', () => {
+		log('debug', `Http2 session${targetHost} connection goaway`);
+	});
+
+	self.http2Connections[targetHost].on('connect', () => {
+		log('debug', `Http2 session${targetHost} connection init`);
+	});
+
+	self.http2Connections[targetHost].on('timeout', () => {
+		log('debug', `Http2 session${targetHost} connection timeout`);
+	});
+
+	self.http2Connections[targetHost].on('close', () => {
+		log('debug', `Http2 session ${targetHost} connection closed`);
+	});
+};
+
+Crawler.prototype._http2request = function _http2request(ropts, options) {
+	const self = this;
+
+	const targetHost = new URL(ropts.uri).origin;
+	ropts.headers = Object.assign(ropts.headers, self.generateHttp2RequestLine(ropts));
+	const requestBody = self.generateHttp2RequestBody(ropts);
```

Comment:
if no need to generate body, let's put the logic outside, say `GET` request. this statement is better to be replaced by `?` operator